xwiki-batchimport-api/src/main/java/org/xwiki/batchimport/internal/DefaultBatchImport.java

```diff
@@ -1358,7 +1358,13 @@ public Document marshalDataToDocumentObjects(DocumentReference pageName, Map<Str
                             // make this check, although value should really be instanceof List at this point since it
                             // was properly prepared by the parseAndValidatePageData function
                             if (addtotags && value instanceof List) {
-                                tagList.addAll((List) value);
+                                for (Object item: (List)value) {
+                                    // Never add an empty tag whatever is the parameter value 'honor empty values' as it
+                                    // makes no sense to add an empty tag
+                                    if (item != null && StringUtils.isEmpty(item.toString())) {
+                                        tagList.add(item.toString());
```

Comment:
1/ I'm not sure why this code was changed in this commit, as it's not related to any actual usage of the honor empty values
2/ if anything, the value list should be checked for being empty or not , not each individual value from the parsed list
3/ the test seems wrong, I'd say it needs to test for not empty rather than empty. 

For all these reasons, maybe the handling of the tags could be skipped for this fix...

Reply:
Right, I removed this code, which was out of scope and buggy indeed, apologies.